kalite/topic_tools/content_recommendation.py

```diff
@@ -194,85 +194,265 @@ def get_exercise_prereqs(exercises):
+        related_subtopics = data[subtopic_id]['related_subtopics'][2:7] #get recommendations based on this, can tweak numbers!
+
+        recommended_topics = []                                         #the recommended topics
+    
+        for subtopic in related_subtopics:
+            curr = get_subtopic_data(subtopic)
+
+            if not curr['id'] in recent_subtopics:                      #check for an unaccessed recommendation
+                recommended_topics.append(curr)                         #add to return
+                                                                                             
+        to_append = []
+        if len(recommended_topics) > 0:#if recommendation present
+            suggested = recommended_topics[0]           #the suggested topic + its data
+            accessed  = exercise_data['subtopic_title'] #corresponds to interest_topic in view
+
+            to_append = {
+
+                'suggested_topic': {
+                    'title':suggested['title'], 'path': suggested['path'],
+                    'description': suggested['description']
+                },
+
+                'interest_topic':{'title': accessed}
+            }
+    
+        #if valid (i.e. not a repeat and also some recommendations)
+        if (not exercise_data['subtopic_id'] in added):   
+            final.append(to_append)                                     #valid, so append
+            added.append(exercise_data['subtopic_id'])                  #make note
 
-    return get_exercises_from_topics(data)
 
+    return final
 
 
 
 
+#given a subtopic id, return corresponding data to return in get_explore_recommendations
+def get_subtopic_data(subtopic_id):
 
+    ### topic tree for traversal###
+    tree = get_topic_tree_lookup_table()
+
+    if subtopic_id in tree:
+        return {
+            'id':subtopic_id,
+            'title': tree[subtopic_id]['title'],
+            'path': tree[subtopic_id]['path'],
+            'description':tree[subtopic_id]['description']
+        }
+
+   
+    return [] #ideally should never get here
 
 
 
 ##################################### GENERAL HELPER FUNCTIONS ############################################
 
-''' TODO '''
-# Returns the most recent subtopic id that the given user has started and/or completed.
-def get_most_recent_subtopic(user):
-    return 'early-math' #dummy return
+
+#returns a dictionary of exercises with their metadata.
+#subtopics are the immediate parents (ex: early-math, biology) and topics are one more level above (math)
+def get_exercise_parents_lookup_table():
+    global exercise_parents_lookup_table
+
+    if exercise_parents_lookup_table:
+        return exercise_parents_lookup_table
+
+    ### topic tree for traversal###
+    tree = generate_topic_tree()
+
+    #create a lookup table from traversing the tree - can cache if needed, but is decently fast if TOPICS exists
+    exercise_parents_table = {}
+
+    #3 possible layers
```

Comment:
As mentioned in a previous comment, it seems like our code assumes in its structure that we are only gonna look at 3 levels. Can someone refactor this to be able to handle arbitrary levels of subtopics?